apps/arena/lib/arena/game_updater.ex

```diff
@@ -1015,7 +1015,17 @@ defmodule Arena.GameUpdater do
     if power_up.aditional_info.status == :AVAILABLE && Player.alive?(player) do
       updated_power_up = put_in(power_up, [:aditional_info, :status], :TAKEN)
 
-      updated_player = update_in(player, [:aditional_info, :power_ups], fn amount -> amount + 1 end)
+      updated_player =
+        update_in(player, [:aditional_info, :power_ups], fn amount -> amount + 1 end)
+        |> update_in([:aditional_info], fn additional_info ->
+          Map.update(additional_info, :health, additional_info.health, fn current_health ->
+            (current_health + additional_info.base_health * power_up.aditional_info.power_up_health_modifier)
+            |> round()
+          end)
+          |> Map.update(:max_health, additional_info.max_health, fn max_health ->
+            (max_health + additional_info.base_health * power_up.aditional_info.power_up_health_modifier) |> round()
+          end)
```

Comment:
These updates are the same. I think we can create a Utils function like `increase_health_by_base_health_fraction(health, base_health, modifier)` What do you think?

Reply:
Even better a more generic `increase_current_by_percentage(current_value, base_value, percentage)`  and in theory you can event pattern match `is_integer(current_value)` to decide if to round or not

Reply:
really nice suggestions! changed

Reply:
great suggestions guys! changed